test/extensions/transport_sockets/tls/cert_validator/default_validator_integration_test.cc

```diff
@@ -74,6 +74,37 @@ TEST_P(SslCertValidatorIntegrationTest, CertValidationFailed) {
   initialize();
   auto conn = makeSslClientConnection({});
   IntegrationCodecClientPtr codec = makeRawHttpConnection(std::move(conn), absl::nullopt);
+  ASSERT_TRUE(codec->connected());
+  test_server_->waitForCounterGe(listenerStatPrefix("ssl.handshake"), 1);
+  EXPECT_EQ(test_server_->counter(listenerStatPrefix("ssl.fail_verify_error"))->value(), 0);
+  codec->close();
+}
+
+// With broken chain set, certificate validation fails without max depth
+TEST_P(SslCertValidatorIntegrationTest, CertValidationFailedWithBrokenChain) {
+  config_helper_.addSslConfig(ConfigHelper::ServerSslOptions()
+                                  .setRsaCert(true)
+                                  .setTlsV13(true)
+                                  .setClientWithIntermediateCert(true)
+                                  .setBrokenChain(true));
+  initialize();
+  auto conn = makeSslClientConnection({});
+  IntegrationCodecClientPtr codec = makeRawHttpConnection(std::move(conn), absl::nullopt);
+  test_server_->waitForCounterGe(listenerStatPrefix("ssl.fail_verify_error"), 1);
+  ASSERT_TRUE(codec->waitForDisconnect());
+}
+
+// With broken chain set, certificate validation fails with max depth
+TEST_P(SslCertValidatorIntegrationTest, CertValidationFailedWithBrokenChainAndDepth) {
+  config_helper_.addSslConfig(ConfigHelper::ServerSslOptions()
+                                  .setRsaCert(true)
+                                  .setTlsV13(true)
+                                  .setClientWithIntermediateCert(true)
+                                  .setBrokenChain(true)
+                                  .setVerifyDepth(1));
```

Comment:
It doesn't seems a useful case to test, it still failed due to broken chain not due to the depth. Or maybe comment more about this test case.

Reply:
Yeah, maybe we need introduce a config option for this feature, otherwise the depth controls nothing.